### DISCUSSION TITLE
Why does the alpinejs ESM module require more than just an import?

### DISCUSSION THREAD

**Discussion by danielniccoli** [alpinejs/alpine#4349] [4. General]
I can add the htmx module to my website by adding it to vite's `main.js` with just

```js
import "htmx.org"
```

For alpinejs, a simple `import "alpinejs"` is not enough, and I must add additional code:

```js
import Alpine from "alpinejs"
window.Alpine = Alpine
Alpine.start()
```

Was this a deliberate design choice? Or was this necessary, because there is something preventing it from having it a simple `import "alpinejs"`, as it does with htmx?

**Comment by SimoTod** [comment]
`window.Alpine = Alpine` is not necessary unless you use Alpine outside of the file/script where you call import.

The cdn autostart,s the module version doesn't so people can do
```js
import Alpine from "alpinejs"

... your code, define Alpine.data, stores, custom magic/directives, etx

Alpine.start()
```

without exposing Alpine on the global object. 

It's a design choice.

**Comment by ekwoka** [comment]
Because then you can't do any setup.

I'd actually argue that HTMX is not good, because of how it self initializes always. It requires more working around how it works than Alpine does. Their extensions are even worse...

HTMX is very much a cool concept, but I think it's engineering is pretty awful. (I say this as a contributor)
